u_parrots.go

```diff
@@ -938,3 +941,367 @@ func removeRC4Ciphers(s []uint16) []uint16 {
 	}
 	return s[:sliceLen]
 }
+
+func FingerprintClientHello(data []byte) (*ClientHelloSpec, error) {
+	m := &clientHelloMsg{raw: data}
```

Comment:
are you preserving the parsing of this because you're expecting we'll needs some pieces from this as well?

Reply:
oic, you use bits that are parsed in there to fill in the extensions and so forth.

Reply:
I think the other part is that it keeps the code as similar to the `unmarshal` code from `crypto/tls` as possible. If you remove the `// utls specific` sections you end up with the exact same code from https://github.com/refraction-networking/utls/blob/43c36d3c1f57546d5cbb05c066df7b5a78686c51/handshake_messages.go#L353

I'm not fully convinced that's worthwhile, but it seemed like it could be useful. fwiw that's the same reason I use the `parse()` closure (so that it can return a bool, same as the original code) :shrug: